quantum/quantum.c

```diff
@@ -757,6 +765,13 @@ void matrix_scan_quantum() {
     backlight_task();
   #endif
 
+  #ifdef RGB_MATRIX_ENABLE
+    // if (rgb_matrix_task_counter == 0)
+    rgb_matrix_task();
+    // rgb_matrix_task_counter = ((rgb_matrix_task_counter + 1) % 5);
+    rgb_matrix_update_pwm_buffers();
```

Comment:
So... this gets called every cycle. `rgb_matrix_update_pwm_buffers` will call `IS31FL3731_update_pwm_buffers` and `IS31FL3731_update_led_control_registers`, both of which are guarded by dirty flags, and only update when things changed.

If we have animations that change something every cycle, updating every cycle will be *slow*. I'm not familiar with the hardware here, but I imagine updating here isn't cheap either (otherwise the dirty flags would be pointless). In Kaleidoscope, we were facing a similar situation, where the keyboard felt laggy (see keyboardio/Kaleidoscope#113 and keyboardio/Kaleidoscope-LEDControl#1 for details and some numbers), so we changed the LED sync to happen only every 16msec or so (still netting us ~60fps), and the results were amazing.

Perhaps it would be worth considering doing something similar here, too?

Reply:
My original code would update backlight effect state using a 20Hz timer/interrupt, but then update the LED driver registers in between matrix scans.

Reply:
I moved it from the timer to the task to avoid conflicts with other timers (1 & 3 are used by audio), and I had played around with the counter that is commented out there a little to lighten the load - I think something like that might work well.